backend/src/Edna/Dashboard/Service.hs

```diff
@@ -0,0 +1,130 @@
+-- | Implementation of Dashboard functionality
+
+module Edna.Dashboard.Service
+  ( -- * Implementation
+    makePrimarySubExperiment
+  , setNameSubExperiment
+  , setIsSuspiciousSubExperiment
+  , deleteSubExperiment
+  , getExperiments
+  , getSubExperiment
+  , getMeasurements
+  ) where
+
+import Universum
+
+import qualified Data.HashSet as HS
+
+import Database.Beam.Backend.SQL.BeamExtensions (unSerial)
+import Database.Beam.Postgres (PgJSON(..))
+import Fmt (fmt, (+|), (|+))
+import Servant.API (NoContent(..))
+
+import qualified Edna.Dashboard.DB.Query as Q
+
+import Edna.Analysis.FourPL (Params4PL(..))
+import Edna.Dashboard.DB.Schema (MeasurementT(..), SubExperimentT(..))
+import Edna.Dashboard.Error (DashboardError(..))
+import Edna.Dashboard.Web.Types
+  (ExperimentResp(..), ExperimentsResp(..), MeasurementResp(..), SubExperimentResp(..))
+import Edna.Logging (logMessage)
+import Edna.Setup (Edna)
+import Edna.Util
+  (CompoundId, IdType(..), ProjectId, SubExperimentId, TargetId, justOrThrow, unSqlId)
+import Edna.Web.Types (WithId(..))
+
+-- | Make given sub-experiment the primary one for its parent experiment.
+-- Previous primary sub-experiment is no longer primary.
+makePrimarySubExperiment :: SubExperimentId ->
+  Edna (WithId 'SubExperimentId SubExperimentResp)
+makePrimarySubExperiment subExpId = do
+  logMessage $ fmt $ "Making sub-experiment " +| subExpId |+ " primary"
```

Comment:
We already log request with responses, looks like no need to duplicate it

Reply:
1. We log them only if logging is `Dev`. And if logging mode is `Prod`, my idea was to log some important actions briefly. And here I tentatively treat only mutating actions (i. e. not getters) as _important_. So in `Prod` mode you will see state changes, but won't see all requests, so the log won't be too big.
2. In `Dev` mode (which I treat as `Debug` mode) this duplication may be not bad because sometimes weird things happen, e. g. an endpoint is called, but no state is changed, and these messages help you understand whether you reached `Q.makePrimarySubExperiment`. Also they are easier to read than full requests because they are written in human language.

That said, I see your point too, so if I didn't convince you that these messages are ok, I'll delete them.

Reply:
>So in Prod mode you will see state changes, but won't see all requests, so the log won't be too big.

As I already wrote, on prod it is also very useful to have full logs

>I'll delete them

Lets leave them to merge faster

Reply:
> As I already wrote, on prod it is also very useful to have full logs

Yes, actually, it's set to Dev in both staging and demo for now: https://github.com/serokell/gemini-infra/pull/10/files
I don't have a real use case for Prod for now, the time will show whether it's useful or not.